agent/consul/leader_connect_test.go

```diff
@@ -653,16 +651,16 @@ func TestLeader_SecondaryCA_IntermediateRefresh(t *testing.T) {
 		}
 		var reply interface{}
 
-		require.NoError(s1.RPC("ConnectCA.ConfigurationSet", args, &reply))
+		require.NoError(t, s1.RPC("ConnectCA.ConfigurationSet", args, &reply))
 	}
 
 	var updatedRoot *structs.CARoot
-	{
+	retry.Run(t, func(r *retry.R) {
```

Comment:
If I understand correctly, what we are waiting on is for the `UpdateConfiguration` to perform the root rotation. 

Do you know why that is not completed synchronously? I thought it should be complete by the time the `RPC` call above returned.

It might be worth it to comment on this line on what async operation we are waiting on.

Reply:
That's a great point. I thought there was some asynchronous handling but I'm not seeing any after looking closer.

I also just noticed this log line in the CI failure:

> TestLeader_SecondaryCA_IntermediateRefresh-node-377.server.connect: pruning old unused root CA: id=`5f:de:2d:bc:45:23:14:dd:48:85:40:e9:58:f8:6e:e2:ad:31:ef:ae`

Maybe the reason why there was only two roots is because the old one was pruned already, but the really strange thing there is that the `caRootPruneInterval` is set to an hour, so we shouldn't have hit the pruning function to begin with.

That aside, maybe the assertion here could be relaxed to check that the active root doesn't match the original, since we just set a new CA configuration.

What do you think?

Reply:
Aha! That's an excellent lead. It looks like `TestLeader_CARootPruning` sets the value to 200 milliseconds, but never resets the old value.  So any test that runs after `TestLeader_CARootPruning` could run into this pruning. That seems like a good thing to fix.

If we fix that pruning interval back to 1 hour, maybe we don't need to make any changes to this test? Relaxing the assertion might also be ok, but I generally like to be confident in the underlying cause of the flakiness. Relaxing the assertion would make the test less flaky, but could hide other bugs.

Reply:
Ah, good find. I reverted the retry and added a commit to reset that value to an hour.

Reply:
From looking at the CI logs, TestLeader_CARootPruning did run first